learn/validator-trust-score.md

```diff
@@ -0,0 +1,227 @@
+
+## The VTS algorithm
+
+The algorithm is a product of three factors: Size, Liveness and Reliability, each ranging from 0 to 1.
+
+$$
+T = S \times L \times R
+$$
+
+The Size factor depends solely on the current distribution of bets. Conversely, the Reliability and Liveness scores are based on their behaviour over the last 9 months, which we believe is a sufficient period to evaluate a validator. Behaviour older than 9 months is excluded from the score calculation.
+
+We only consider completed epochs, not the currently active epoch. Therefore, the score is not a live value. It can have a delay of +12 hours (an epoch last 12 hours).
+
+Before moving forward, we will define $m$, which is the number of epochs to consider knowing that the window duration is 9 months.
+
+<details>
+
+<summary children:mx-0>
+
+Calculation of $m$
+
+</summary>
+
+$$
+m = \frac{\text{window\_duration\_ms}}{\text{epoch\_duration}}
+$$
+
+$$
+$\text{window\_duration\_ms} = 9 \times 30 \times 24 \times 60 \times 60 \times 1000$.
+$$
+
+$$
+$\text{epoch\_duration} = \text{block\_duration} \times \text{blocks\_per\_epoch}
+$$
+
+  - Block duration and blocks per epoch are constants from the [policy](https://github.com/nimiq/core-rs-albatross/blob/albatross/primitives/src/policy.rs)
+
+</details>
+
+Let's look at how each parameter is calculated:
+
+### Size
+
+The size factor penalises validators who control too much of the network's share.
+
+We define the size $s$ of a validator as the fraction of the total stake it controls. The size factor is calculated as
+
+$$
+S = \max \left( 0 , 1 - \left( \frac{s}{t} \right)^{k} \right), \quad \text{being } t = 0.25 \text{ and } k=4
+$$
+
+Where $t$ is the threshold and $k$ is the slope of the curve
+
+<figure mb-16>
+
+<iframe src="https://www.desmos.com/calculator/89yvaeqxff?embed" aspect-video frameborder="0" allowfullscreen></iframe>
+
+<figcaption>
+
+Graph of the size factor. The x-axis represents the size of the validator, and the y-axis represents the size factor.
+
+</figcaption>
+
+</figure>
+
+For example:
+
+- A validator with a size of 0.1 would have a size factor of $0.974$.
+- A validator with a size of 0.15 would have a size factor of $0.87$.
+
+<Callout type="info" no-title>
+
+Due to technical limitations, we can currently only calculate the size of validators active in the current epoch. We cannot calculate the score of a validator in a given timestamp.
+
+</Callout>
+
+### Liveliness
+
+The liveness factor penalises validators that are not selected to produce blocks and encourages validators to be selected.
+
+The score is a moving average of the liveness score for each epoch. First, we calculate the liveness($l_i$) for each epoch.
+
+<Callout type="info" no-title>
+
+Take the number of epochs in which the validator was selected and divide by the number of epochs.
+
+</Callout>
+
+$$
+[ l_i = \frac{\text{produced}_i}{\text{likelihood}_i}, \quad \text{for } i = 0, 1, 2, \ldots, m-1 ]
```

Comment:
Maybe remove the square brackets?
Seems like they are not needed here since it's not a list.

In any case it should be `l_i = \frac{\text{produced}_i}{\text{likelihood}_i}, \quad \text{for } i = [ 0, 1, 2, \ldots, m-1 ]`

Reply:
The [] represent a list of elemnets.

i will ask chat gpt whatever he says i would put

Reply:
they will be removed